api/src/main/java/org/ccci/idm/user/DefaultUserManager.java

```diff
@@ -155,6 +164,11 @@ public void updateUser(final User user, final User.Attr... attrs) throws DaoExce
         // update the user object
         this.userDao.update(original, user, attrs);
 
+        // add password to history (if you have password and caller intends to set)
+        if(StringUtils.hasText(user.getPassword()) && Iterables.contains(Arrays.asList(attrs), User.Attr.PASSWORD)) {
```

Comment:
something I just tried out that makes the code more readable:

`FluentIterable.of(attrs).contains(User.Attr.PASSWORD)`

this isn't blocking, just a more readable way to check for the Password attr set